TribalWars/src/Infrastructure/Repository/BuildingsRepository.cs

```diff
@@ -0,0 +1,24 @@
+﻿using ApplicationCore.Entities;
+using ApplicationCore.Interfaces;
+using Infrastructure.Data;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Infrastructure.Repository
+{
+    public class BuildingsRepository : BaseRepository, IBuildingsRepository
+    {
+        public BuildingsRepository(ApplicationDbContext context) : base(context)
+        {
+        }
+
+        public Building SelectedBuilding(int idBuilding)
```

Comment:
This method doesn't tell the developer anything. As it is a getter method, we would name it like `Get<Entity>By<Params>` so in this case `GetBuildingById`